guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,868 @@
+`JitteredResize`, let's use the deterministic `keras_cv.layers.Resizing()`
+layer.
+"""
+
+inference_resizing = keras_cv.layers.Resizing(
+    640, 640, bounding_box_format="xywh", pad_to_aspect_ratio=True
+)
+eval_ds = eval_ds.map(inference_resizing, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Due to the fact that the resize operation differs between the train dataset,
+which uses `JitteredResize()` to resize images, and the inference dataset which
+uses `layers.Resizing(pad_to_aspect_ratio=True)`, it is good practice to
+visualize both datasets:
+"""
+
+visualization.visualize_dataset(eval_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2)
+
+"""
+Finally, let's unpackage our inputs from the preprocessing dictionary, and
+prepare to feed the inputs into our model.  In order to be TPU compatible,
+bounding box Tensors need to be `Dense` instead of `Ragged`.  If training on
+GPU, you can omit the `bounding_box.to_dense()` call.  If ommitted,
+the KerasCV RetinaNet
+label encoder will automatically correctly encode Ragged training targets.
+"""
+
+def dict_to_tuple(inputs):
+    return inputs["images"], bounding_box.to_dense(
+        inputs["bounding_boxes"], max_boxes=32
+    )
+
+train_ds = train_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete!
+We can now move on to model creation and training.
+
+### Model creation
+
+We'll use the KerasCV API to construct a RetinaNet model.
+In this tutorial we use a pretrained ResNet50 backbone, initializing the
+weights to weights produced by training a classification model on the imagenet
+dataset.
+
+KerasCV makes it easy to construct a `RetinaNet` with any of the KerasCV
+backbones.  Simply use one of the presets for the architecture you'd like!
+
+For example:
+"""
+
+model = keras_cv.models.RetinaNet.from_preset(
+    "resnet50_imagenet",
+    # number of classes to be used in box classification
+    num_classes=len(class_mapping) + 1,
+    # For more info on supported bounding box formats, visit
+    # https://keras.io/api/keras_cv/bounding_box/
+    bounding_box_format="xywh",
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts
+tuples of dense image Tensors and bounding box dictionaries to `fit()` and
+`train_on_batch()`
+
+This matches what we have constructed in our input pipeline above.
+
+**Note: be sure to freeze BatchNormalization layers when training a RetinaNet.***
+
+And important but easy to miss step in training your own object detection model
+is freezing the `BatchNormalization` layers in your backbone.
+We felt it would be confusing to do this automatically,
+so instead we recommend you do it in your training loops.
+"""
+
+for layer in model.backbone.layers:
```

Comment:
Should this be a method on `RetinaNet`? Seems very handy

```python
model.freeze_batchnorm_layers()
model.unfreeze_batchfrom_layers()
```

Reply:
maybe in a follow up, for now lets leave it.

BatchNorm is something I think Keras in general could handle a bit better - (sync batch norm, normal batch norm, freezing, unfreezing)